torch/_inductor/utils.py

```diff
@@ -727,6 +727,22 @@ def instance_descriptor(
     )
 
 
+_registered_caches: List[Any] = []
+
+
+def clear_on_fresh_inductor_cache(obj: Any):
+    """
+    Use this decorator to register any caches that should be cache_clear'd
+    with fresh_inductor_cache().
+    """
+    if not hasattr(obj, "cache_clear") or not callable(obj.cache_clear):
+        raise AttributeError(f"{obj} does not have a cache_clear method")
```

Comment:
do we need this? dont we only add this to "correct" functions?

Reply:
Yes, it's not strictly needed. I'm just trying to give a more readable error message if someone slaps this decorator on a function incorrectly. Ideally, I could get mypy to complain if used incorrectly, but AFAIK there's no way to do it robustly because Python.